use node-gyp

themitchy opened the pull request:
cc @bnoordhuis 

Really just like the title says. No more node-waf so use node-gyp.

themitchy committed:
use node-gyp

bnoordhuis commented on Makefile:
```diff
@@ -22,8 +22,8 @@ build: bin
 	# remove --dest-cpu ia32 to compile in x64 on 64bit host
-	node-waf configure --dest-cpu ia32
-	node-waf build
+	node-gyp configure --dest-cpu ia32
```
Maybe replace `ia32` with `$ARCH`?

themitchy replied:
--dest-cpu x64 results in:
gyp: x64 not found

bnoordhuis replied:
Oh, it's `--dest-cpu=${ARCH}`.  I guess you get the same issue with `--dest-cpu ia32`, node-gyp doesn't like the space.

themitchy replied:
good catch!

bnoordhuis commented:
The commit log could elaborate a little more.  Something like:

```
build: use node-gyp, remove node-waf

node-waf was deprecated in node.js v0.8 and removed in v0.10.
We only support v0.10 so drop support for node-waf.
```

themitchy committed:
Use detected architecture.

The flag `--dest-cpu ia32` would have been throwing the same error.
Correct usage is `--dest-cpu=...`
